pkg/handler/snapshots.go

```diff
@@ -170,3 +189,206 @@ func (sh *SnapshotHandler) getRepoConfigurationFile(c echo.Context) error {
+	}
+	snapshotUUIDs = body.UUIDs
+
+	repoSnaps, err := sh.isDeleteAllowed(c, orgID, repoUUID, snapshotUUIDs...)
+	if err != nil {
+		var errResp ce.ErrorResponse
+		if errors.As(err, &errResp) {
+			return errResp
+		}
+		return ce.NewErrorResponse(http.StatusBadRequest, "Error deleting snapshots", err.Error())
+	}
+
+	responses := make([]api.SnapshotResponse, len(snapshotUUIDs))
+	hasErr := false
+	errs := make([]error, len(snapshotUUIDs))
+	for i := range snapshotUUIDs {
+		snap, err := sh.DaoRegistry.Snapshot.Fetch(c.Request().Context(), snapshotUUIDs[i])
+		responses[i] = snap
+		if err != nil {
+			hasErr = true
+			errs[i] = err
+			continue
+		}
+		err = sh.isSnapInRepo(repoSnaps, snap.UUID)
+		if err != nil {
+			hasErr = true
+			errs[i] = err
+			continue
+		}
+	}
+	if hasErr {
+		return ce.NewErrorResponseFromError("Error deleting snapshots", errs...)
+	}
+
+	errs = sh.DaoRegistry.Snapshot.BulkDelete(c.Request().Context(), snapshotUUIDs)
+	if len(errs) > 0 {
+		return ce.NewErrorResponseFromError("Error deleting snapshots", errs...)
+	}
+
+	enqueueErr := sh.enqueueDeleteSnapshotsTask(c, orgID, repoUUID, snapshotUUIDs...)
+	if enqueueErr != nil {
+		hasErr = false
+		errs = make([]error, len(snapshotUUIDs))
+		for i := range snapshotUUIDs {
+			err = sh.DaoRegistry.Snapshot.ClearDeletedAt(c.Request().Context(), snapshotUUIDs[i])
+			if err != nil {
+				hasErr = true
+				errs[i] = err
+			}
+		}
+		if hasErr {
+			return ce.NewErrorResponseFromError("Error clearing snapshot deleted_at fields", errs...)
+		}
+		return ce.NewErrorResponse(ce.HttpCodeForDaoError(enqueueErr), "Error enqueueing task", enqueueErr.Error())
+	}
+
+	return c.NoContent(http.StatusNoContent)
+}
+
+func (sh *SnapshotHandler) enqueueDeleteSnapshotsTask(c echo.Context, orgID, repoUUID string, snapshotUUIDs ...string) error {
+	accountID, _ := getAccountIdOrgId(c)
+	payload := payloads.DeleteSnapshotsPayload{RepoUUID: repoUUID, SnapshotsUUIDs: snapshotUUIDs}
+	task := queue.Task{
+		Typename:   config.DeleteSnapshotsTask,
+		Payload:    payload,
+		OrgId:      orgID,
+		AccountId:  accountID,
+		ObjectUUID: utils.Ptr(repoUUID),
+		ObjectType: utils.Ptr(config.ObjectTypeRepository),
+		RequestID:  c.Response().Header().Get(config.HeaderRequestId),
+	}
+
+	taskID, err := sh.TaskClient.Enqueue(task)
+	if err != nil {
+		logger := tasks.LogForTask(taskID.String(), task.Typename, task.RequestID)
+		logger.Error().Msg("error enqueuing task")
+		return err
+	}
+
+	return nil
+}
+
+func (sh *SnapshotHandler) isDeleteInProgress(c echo.Context, orgID, repoUUID string) error {
+	inProgressTasks, err := sh.DaoRegistry.TaskInfo.
+		FetchActiveTasks(c.Request().Context(), orgID, repoUUID, config.DeleteRepositorySnapshotsTask, config.DeleteSnapshotsTask)
+
+	if err != nil {
+		return ce.NewErrorResponse(ce.HttpCodeForDaoError(err), "Error fetching delete repository snapshots task", err.Error())
```

Comment:
Seeing this error if deleting with an invalid repository UUID: 

`Error fetching delete repository snapshots task, detail=ERROR: invalid input syntax for type uuid: \"not-a-valid-repository\" (SQLSTATE 22P02) \n`

I think I broke this in an earlier PR, but the FetchActiveTasks method should be calling the UuidifyString function on the uuid passed to it.

Reply:
Added the `UuidifyString()` to `FetchActiveTasks()` and a repo config check when deleting a snapshot, so this should  never happen again.